feat(linter/plugins): move custom JS plugin config to `jsPlugins`

overlookmotel opened the pull request:
Fixes #12432.

JS plugins were previously configured by adding them to `plugins` array in config file. Instead, move them to a separate `jsPlugins` array.

Before:

```json
{
  "plugins": ["oxc", "react", "./custom-plugin.js"],
}
```

After:

```json
{
  "plugins": ["oxc", "react"],
  "jsPlugins": ["./custom-plugin.js"],
}
```

overlookmotel commented on crates/oxc_linter/src/config/oxlintrc.rs:
```diff
@@ -191,14 +194,24 @@ impl Oxlintrc {
+            (None, other_plugins) => other_plugins,
+        };
+
+        let external_plugins = match (&self.external_plugins, &other.external_plugins) {
+            (Some(self_external), Some(other_external)) => {
+                Some(self_external.iter().chain(other_external).cloned().collect())
+            }
+            (Some(self_external), None) => Some(self_external.clone()),
+            (None, Some(other_external)) => Some(other_external.clone()),
+            (None, None) => None,
```
Again with this nonsense??

overlookmotel commented on crates/oxc_linter/src/config/config_store.rs:
```diff
@@ -156,37 +156,32 @@ impl Config {
-                for p in &override_plugins.external {
-                    plugins.external.insert(p.clone());
-                }
+            if let Some(override_plugins) = override_config.plugins {
+                plugins |= override_plugins;
```
I *think* this is OK. We're not storing external plugins in `ResolvedLinterState`, but I don't think we need to, because all that matters at this point is which external *rules* are enabled.

camc314 commented:
### Merge activity

* **Sep 26, 10:05 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(linter/plugins): move custom JS plugin config to `jsPlugins` (#14133)

Fixes #12432.

JS plugins were previously configured by adding them to `plugins` array in config file. Instead, move them to a separate `jsPlugins` array.

Before:

```json
{
  "plugins": ["oxc", "react", "./custom-plugin.js"],
}
```

After:

```json
{
  "plugins": ["oxc", "react"],
  "jsPlugins": ["./custom-plugin.js"],
}
```
